### PR TITLE
ARROW-15934: [Python] Expose write_batch_size in python

### PR DESCRIPTION
**Pull request by tachyonwill**
Already exposed in C++. Useful when dealing columns of large strings.

### CODE REVIEW

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks for the PR! 

Can you also add a test for it? At least testing that the keyword can be passed, ideally also checking something about the resulting file that is influenced by that (although I don't know if our python APIs make that possible)

File: python/pyarrow/parquet.py [hunk 1]
```diff
@@ -605,6 +605,9 @@ def _sanitize_table(table, new_schema, flavor):
     If None, no encryption will be done.
     The encryption properties can be created using:
     ``CryptoFactory.file_encryption_properties()``.
+write_batch_size : int, default None
+    Number of values to write to a page at a time. If None, use the default of
+    1024.
```
[the remark this answers is not in the document]
Comment (tachyonwill) [reply python/pyarrow/parquet.py:610]: The default is 1024: https://github.com/apache/arrow/blob/3bf061783f4e1ab447d2eb0f487c0c4fce6d5b15/cpp/src/parquet/properties.h#L96

The way `data_page_size` and `write_batch_size` work is `write_batch_size` values are written, then the size of the page is checked against `data_page_size` and if `data_page_size` is exceeded, we start a new page. Normally, 1024 is fine for the `write_batch_size` but if the values are really big(big strings) or  `data_page_size` really small, then a smaller `write_batch_size` is needed to keep the page sizes close to `data_page_size` .

[the remark this answers is not in the document]
Comment (jorisvandenbossche) [reply python/pyarrow/parquet.py:610]: Ah, yes, sorry was looking at the wrong line (it seems that the other constant I pointed to is never used anywhere).

Thanks for the explanation how those two arguments interact. Would it be worth adding something about that to the docstring?

[the remark this answers is not in the document]
Comment (tachyonwill) [reply python/pyarrow/parquet.py]: Added more commentary in the docstring, but I am intentionally trying to be somewhat vague here and avoid exposing too many implementation details. I think we might want to change the way we write parquet pages in the future. For example, I suspect, but haven't investigated, that the current logic does not work correctly for V2 data pages.

**Comment by jorisvandenbossche** [comment]
> Added more commentary in the docstring, but I am intentionally trying to be somewhat vague here 

Thanks for the update, sounds good!

> Added a test, however I don't think there is a good way to inspect the resulting pages sizes programmatically so this is really just a test that the keywords are accepted.

**Comment by jorisvandenbossche** [comment]
I just moved the test from `test_basic.py` (`test_parquet_writer.py` currently has tests specifically for the ParquetWriter API, while this is a test for a generic keyword, and eg the `data_page_size` keyword is also tested in `test_basic.py`)

**Review by jorisvandenbossche — APPROVED** [review]

**Pull request closed without merging**